sql/moz-fx-data-shared-prod/firefox_desktop_derived/newtab_clients_daily_v2/query.sql

```diff
@@ -0,0 +1,403 @@
+    MAX(
+      IF(
+        event_category = 'topsites'
+        AND event_name IN ('click', 'dismiss')
+        AND SAFE_CAST(mozfun.map.get_key(event_details, 'is_sponsored') AS BOOLEAN),
+        TRUE,
+        FALSE
+      )
+    ) AS is_sponsored_topsites_interaction,
+    MAX(
+      IF(
+        event_category = 'topsites'
+        AND event_name IN ('click')
+        AND SAFE_CAST(mozfun.map.get_key(event_details, 'is_sponsored') AS BOOLEAN),
+        TRUE,
+        FALSE
+      )
+    ) AS is_sponsored_topsites_click,
+    MAX(
+      IF(
+        event_category = 'topsites'
+        AND event_name IN ('impression')
+        AND SAFE_CAST(mozfun.map.get_key(event_details, 'is_sponsored') AS BOOLEAN),
+        TRUE,
+        FALSE
+      )
+    ) AS is_sponsored_topsites_impression,
+    COUNTIF(event_category = 'topsites' AND event_name IN ('click')) AS any_topsites_click_count,
+    COUNTIF(
+      event_category = 'topsites'
+      AND event_name IN ('impression')
+    ) AS any_topsites_impression_count,
+    COUNTIF(
+      event_category = 'topsites'
+      AND event_name IN ('click')
+      AND NOT SAFE_CAST(mozfun.map.get_key(event_details, 'is_sponsored') AS BOOLEAN)
+    ) AS organic_topsites_click_count,
+    COUNTIF(
+      event_category = 'topsites'
+      AND event_name IN ('impression')
+      AND NOT SAFE_CAST(mozfun.map.get_key(event_details, 'is_sponsored') AS BOOLEAN)
+    ) AS organic_topsites_impression_count,
+    COUNTIF(
+      event_category = 'topsites'
+      AND event_name IN ('click')
+      AND SAFE_CAST(mozfun.map.get_key(event_details, 'is_sponsored') AS BOOLEAN)
+    ) AS sponsored_topsites_click_count,
+    COUNTIF(
+      event_category = 'topsites'
+      AND event_name IN ('impression')
+      AND SAFE_CAST(mozfun.map.get_key(event_details, 'is_sponsored') AS BOOLEAN)
+    ) AS sponsored_topsites_impression_count,
+    MAX(
+      IF(
+        event_category = 'newtab'
+        AND event_name IN (
+          'weather_change_display',
+          'weather_open_url_provider',
+          'weather_location_selected'
+        ),
+        TRUE,
+        FALSE
+      )
+    ) AS is_widget_interaction,
+    MAX(
+      IF(
+        event_category = 'newtab'
+        AND event_name IN (
+          'wallpaper_click',
+          'wallpaper_category_click',
+          'wallpaper_highlight_cta_click',
+          'wallpaper_highlight_dismissed'
+        ),
+        TRUE,
+        FALSE
+      )
+    ) AS is_wallpaper_interaction,
+    MAX(
+      IF(
+        event_category = 'newtab'
+        AND event_name IN (
+          'topic_selection_open',
+          'topic_selection_dismiss',
+          'topic_selection_topics_saved',
+          'sections_block_section',
+          'sections_follow_section',
+          'sections_unblock_section',
+          'sections_unfollow_section',
+          'inline_selection_click'
+        ),
+        TRUE,
+        FALSE
+      )
+    ) AS is_other_interaction,
```

Comment:
`is_wallpaper_interaction` and `is_other_interaction`, as defined here all fit into the `other_engagement` category. that is to say, wallpaper interactions are also considered 'other' interactions.

I see the doc was confusing in this part. I edited it for clarity, and commented.

Reply:
I will fix this in the code and will include the wallpaper interactions into `is_other_interaction` as well, so that we can use:
- `is_wallpaper_interaction` for only wallpaper
- `is_other_interaction` for other engagement (will include all the other including wallpaper)